rust/operator-binary/src/main.rs

```diff
@@ -88,23 +99,65 @@ async fn main() -> anyhow::Result<()> {
                 );
             });
 
-            let init_controller = Controller::new(
-                watch_namespace.get_api::<Init>(&client),
+            let airflow_db_controller_builder = Controller::new(
+                watch_namespace.get_api::<AirflowDB>(&client),
                 ListParams::default(),
-            )
-            .shutdown_on_signal()
-            .run(
-                init_controller::reconcile_init,
-                init_controller::error_policy,
-                Context::new(init_controller::Ctx {
-                    client: client.clone(),
-                }),
-            )
-            .map(|res| {
-                report_controller_reconciled(&client, "inits.command.airflow.stackable.tech", &res);
-            });
+            );
+
+            let airflow_db_store1 = airflow_db_controller_builder.store();
+            let airflow_db_store2 = airflow_db_controller_builder.store();
+            let airflow_db_controller = airflow_db_controller_builder
+                .shutdown_on_signal()
+                .watches(
+                    watch_namespace.get_api::<Secret>(&client),
+                    ListParams::default(),
+                    move |secret| {
+                        airflow_db_store1
+                            .state()
+                            .into_iter()
+                            .filter(move |airflow_db| {
+                                if let Some(n) = &secret.metadata.name {
+                                    &airflow_db.spec.credentials_secret == n
+                                } else {
+                                    false
+                                }
+                            })
+                            .map(|airflow_db| ObjectRef::from_obj(&*airflow_db))
+                    },
+                )
+                // We have to watch jobs so we can react to finished init jobs
+                // and update our status accordingly
+                .watches(
+                    watch_namespace.get_api::<Job>(&client),
+                    ListParams::default(),
+                    move |job| {
+                        airflow_db_store2
+                            .state()
+                            .into_iter()
+                            .filter(move |airflow_db| {
+                                airflow_db.metadata.namespace.as_ref().unwrap()
+                                    == job.metadata.namespace.as_ref().unwrap()
+                                    && &airflow_db.job_name() == job.metadata.name.as_ref().unwrap()
```

Comment:
```suggestion
                                airflow_db.name() == job.name()
                                    && airflow_db.namespace() == job.namespace()
```
You have to `use stackable_operator::kube::ResourceExt;` for this to work.
This is generic to all the k8s objects out there i can highly recommend these functions

Reply:
Nice!